### PR TITLE
[Merged by Bors] - chore: squeeze some non-terminal simps

### PR DESCRIPTION
**Pull request by adomani**
This PR accompanies #11246, squeezing some non-terminal `simp`s highlighted by the linter until I decided to stop!

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mattrobball** [comment]
!bench

File: Mathlib/Data/Sym/Basic.lean [hunk 1]
```diff
@@ -688,7 +688,8 @@ theorem encode_decode [DecidableEq α] (s : Sum (Sym (Option α) n) (Sym α n.su
     · refine' congr_arg Sum.inr _
       refine' map_injective (Option.some_injective _) _ _
       refine' Eq.trans _ (Eq.trans (SymOptionSuccEquiv.decode (Sum.inr s)).attach_map_coe _)
```
Comment (mattrobball) [inline Mathlib/Data/Sym/Basic.lean:690]: Can you fit a `<;> simp` onto the previous line?

**Review by mattrobball — APPROVED** [review]
Looks good to me! Thanks.

**Comment by mattrobball** [comment]
Can you address the small comment first? 

bors delegate+

File: Mathlib/Data/Sym/Basic.lean (hunk 1 above)
Comment (adomani) [reply Mathlib/Data/Sym/Basic.lean]: I could, explointing that Lean accepts `.trans` for `Eq.trans`!

**Pull request closed without merging**